include/dali/core/geom/transform.h

```diff
@@ -92,6 +92,39 @@ constexpr mat4 shear(mat3x2 shear) {
   }};
 }
 
+template <int out_n, int in_n>
+DALI_HOST_DEV
+constexpr vec<out_n> affine(const mat<out_n, in_n + 1> &transform, const vec<in_n> &v) {
+  vec<out_n> out = {};
+  for (int i = 0; i < out_n; i++) {
+    float sum = transform(i, in_n);
+    for (int j = 0; j < in_n; j++) {
+      sum += transform(i, j) * v[j];
+    }
+    out[i] = sum;
```

Comment:
Add some comment that this is intended and assigning to `out[i]` directly is slower.

Reply:
Will do in another PR.